src/odemis/gui/comp/overlay/ellipse.py

```diff
@@ -74,10 +84,36 @@ def move_to(self, pos):
             self._points[i] += shift
 
     def refresh(self):
-        """Refresh the shape's points VA after copy or move_to operation."""
+        """Refresh the shape's points VA after copy, move_to or restore_state operation."""
         self._phys_to_view()
         self.points.value = self._points
 
+    def save_state(self):
+        """Save the current state of the shape."""
+        state = {
+            EllipseStateKeys.P_POINT1: self.p_point1,
+            EllipseStateKeys.P_POINT2: self.p_point2,
+            EllipseStateKeys.P_POINT3: self.p_point3,
+            EllipseStateKeys.P_POINT4: self.p_point4,
+            EllipseStateKeys.POINTS_PRIVATE: self._points.copy()
```

Comment:
`._points` is computed based on the 4 p_points, so I'd argue it doesn't need to be stored as a state.